### PR TITLE
Add platform developer section to tasks index

### LINKED ISSUE #1007: Organize tasks under personas
**What would you like to be added**:

/kind documentation

Organize the tasks documentation under 3 personas. Maybe this can be implemented using [hugo taxonomies](https://gohugo.io/content-management/taxonomies/)

- Application Developer (previously known as Batch User)
- Batch Ops (previously known as Batch Admin)
- Platform Developer (new)

Caveats:
- In smaller organisations, Batch Ops and Platform developer could be the same person/team.
- Some tasks might apply to more than one persona. For example, an application developer might want to run their jobs through python. Whereas the platform developer might want to develop some tools for Kueue using python as well. Here is where tags could help to list one tutorial under multiple personas.

**Why is this needed**:

As the number of tasks grows, it's useful to have some separation.

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [x] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by BinL233**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:

Add Platform Developer section in `docs/tasks`.

#### Which issue(s) this PR fixes:

Fixes #1007

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: site/content/en/docs/tasks/_index.md [hunk 1]
```diff
@@ -34,3 +34,9 @@ batch user is a researcher, AI/ML engineer, data scientist, among others.
 - As a batch user, you can learn how to [run a Kueue managed Kubeflow MPIJob](/docs/tasks/run_mpi_jobs).
 - As a batch user, you can learn how to [run a Kueue managed KubeRay RayJob](/docs/tasks/run_rayjobs).
 - As a batch developer user, you can learn how to [submit Kueue jobs from Python](/docs/tasks/run_python_jobs).
+
+### Platform Developer
+
+A _Platform Developer_ contributes to Kueue, including code, documentation and etc..
```
Comment (alculquicondor) [inline site/content/en/docs/tasks/_index.md:40]:
```suggestion
A _Platform Developer_ integrates Kueue with other software and/or contributes to Kueue.
```

File: site/content/en/docs/tasks/integrate_a_custom_job.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ There are two options for integrating a Job-like CRD with Kueue:
 - As part of the Kueue repository
 - Writing an external controller
 
-This guide focuses on the first approach.
+This guide is for [platform developer](/docs/tasks#platform-developer) and focuses on the first approach.
```
Comment (alculquicondor) [inline site/content/en/docs/tasks/integrate_a_custom_job.md:15]:
```suggestion
This guide is for [platform developers](/docs/tasks#platform-developer) and focuses on the first approach.
```

**Review by alculquicondor — COMMENTED** [review]
Did you look into setting taxonomies? https://gohugo.io/content-management/taxonomies/

File: site/content/en/docs/tasks/_index.md [hunk 3]
```diff
@@ -17,26 +17,30 @@ personas such as _batch administrators_ and _batch users_.
...
+- [Run a Kueue managed batch/Job](/docs/tasks/run_jobs).
+- [Run a Kueue managed Flux MiniCluster](/docs/tasks/run_flux_minicluster).
+- [Run a Kueue managed Kubeflow MPIJob](/docs/tasks/run_mpi_jobs).
+- [Run a Kueue managed KubeRay RayJob](/docs/tasks/run_rayjobs).
+- [Submit Kueue jobs from Python](/docs/tasks/run_python_jobs).
+
+### Platform developer
 
-### Platform Developer
+A _Platform developer_ integrates Kueue with other software and/or contributes to Kueue.
```
Comment (alculquicondor) [inline site/content/en/docs/tasks/_index.md:41]:
```suggestion
A _platform developer_ integrates Kueue with other software and/or contributes to Kueue.
```

**Review by tenzen-y — COMMENTED** [review]
/lgtm

**Pull request merged**